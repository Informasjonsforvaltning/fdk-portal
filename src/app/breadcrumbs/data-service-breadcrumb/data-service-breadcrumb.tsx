import React from 'react';
import PropTypes from 'prop-types';
import _ from 'lodash';
import { resolve } from 'react-resolver';
import {
  searchDataServices,
  extractDataServices
} from '../../../api/search-fulltext-api/dataservices';
import { getTranslateText } from '../../../lib/translateText';

const memoizedSearchDataServices = _.memoize(searchDataServices);

export const PureDataServiceBreadcrumb = ({ dataServiceItem }: any) => (
  <span>{getTranslateText(dataServiceItem?.title)}</span>
);

const mapProps = {
  dataServiceItem: async (props: any) => {
    const dataServices = await memoizedSearchDataServices({
      q: props.match.params.id
    }).then(extractDataServices);

    return Array.isArray(dataServices) ? dataServices[0] : null;
  }
};

PureDataServiceBreadcrumb.defaultProps = {
  dataServiceItem: null
};

PureDataServiceBreadcrumb.propTypes = {
  dataServiceItem: PropTypes.object
};

export const DataServiceBreadcrumb = resolve(mapProps)(
  PureDataServiceBreadcrumb
);
