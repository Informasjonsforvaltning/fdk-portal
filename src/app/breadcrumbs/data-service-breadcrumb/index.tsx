import React, { memo, FC } from 'react';
import { compose } from 'redux';

import { getTranslateText as translate } from '../../../lib/translateText';
import translations from '../../../lib/localization';

import withDataService, {
  Props as DataServiceProps
} from '../../../components/with-data-service';

interface Props extends DataServiceProps {}

const DataServiceBreadcrumb: FC<Props> = ({ dataService }) => (
  <span>
    {translate(dataService?.title ?? translations.breadcrumb.notFound)}
  </span>
);

export default compose<FC>(memo, withDataService)(DataServiceBreadcrumb);
