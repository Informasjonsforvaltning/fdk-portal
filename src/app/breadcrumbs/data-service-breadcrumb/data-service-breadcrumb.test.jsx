import React from 'react';
import { shallow } from 'enzyme';
import { PureDataServiceBreadcrumb } from './data-service-breadcrumb';

test('should render PureDataServiceBreadcrumb with data service active correctly', () => {
  const defaultProps = {
    dataServiceItem: {
      title: 'Data service title'
    }
  };
  const wrapper = shallow(<PureDataServiceBreadcrumb {...defaultProps} />);
  expect(wrapper).toMatchSnapshot();
});

test('should render PureDataServiceBreadcrumb correctly with no dataServiceItem', () => {
  const defaultProps = {};
  const wrapper = shallow(<PureDataServiceBreadcrumb {...defaultProps} />);
  expect(wrapper).toMatchSnapshot();
});
