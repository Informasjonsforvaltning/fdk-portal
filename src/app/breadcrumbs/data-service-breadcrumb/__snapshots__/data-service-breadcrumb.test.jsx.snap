// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render PureDataServiceBreadcrumb correctly with no dataServiceItem 1`] = `<span />`;

exports[`should render PureDataServiceBreadcrumb with data service active correctly 1`] = `
<span>
  Data service title
</span>
`;
